processor/fanoutconnector.go

```diff
@@ -57,10 +81,73 @@ var _ TraceProcessor = (*traceFanOutConnector)(nil)
 // ConsumeTraceData exports the span data to all trace consumers wrapped by the current one.
 func (tfc traceFanOutConnector) ConsumeTraceData(ctx context.Context, td consumerdata.TraceData) error {
 	var errs []error
-	for _, tc := range tfc {
-		if err := tc.ConsumeTraceData(ctx, td); err != nil {
+
+	// Fan out to first len-1 consumers.
+	for i := 0; i < len(tfc)-1; i++ {
+		// Create a clone of data. We need to clone because consumers may modify the data.
+		clone, err := cloneTraceData(&td)
+		if err != nil {
 			errs = append(errs, err)
+			break
+		} else {
+			if err := tfc[i].ConsumeTraceData(ctx, *clone); err != nil {
+				errs = append(errs, err)
+			}
 		}
 	}
+
+	if len(tfc) > 0 {
+		// Give the original data to the last consumer.
+		lastTc := tfc[len(tfc)-1]
+		if err := lastTc.ConsumeTraceData(ctx, td); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
 	return oterr.CombineErrors(errs)
 }
+
+func cloneTraceData(td *consumerdata.TraceData) (*consumerdata.TraceData, error) {
+	clone := &consumerdata.TraceData{
+		SourceFormat: td.SourceFormat,
+		Node:         proto.Clone(td.Node).(*commonpb.Node),
+		Resource:     proto.Clone(td.Resource).(*resourcepb.Resource),
+	}
+
+	if td.Spans != nil {
+		clone.Spans = make([]*tracepb.Span, 0, len(td.Spans))
+
+		for _, span := range td.Spans {
+			if span != nil {
```

Comment:
I think proto.Clone takes care of this case: https://github.com/golang/protobuf/blob/master/proto/clone.go#L47.